### PR TITLE
Fix crash when trying to alter temp table column type.

### PR DESCRIPTION
**Pull request by Sairakan**
### Description

In cases where a temp table column is being altered so that only the size changes (e.g. char(20) -> char(5)), the code path needs to drop the pg_depend entry from the normal catalog table (to be later recreated using the new target type). However, for ENR temp tables all catalog tuples are stored in currentQueryEnv and not actually in the catalogs themselves. Thus, the code would not be able to find the tuple for deletion, leading to an assertion error. Fix the issue by using ENRDropTuple() instead of CatalogTupleDelete() so that the code drops the tuple in the correct place.

### Issues Resolved

BABEL-5273

### Test Scenarios Covered ###
* **Use case based -**
Tests altering temp table column types
```
ALTER TABLE #t1 ALTER COLUMN b char(5)
```

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**
Try to alter a temp table column type to a smaller type when there is already data in the table larger than the new type would allow (raises an error)
```
ALTER TABLE #t1 ALTER COLUMN b char(4)
GO
~~ERROR (Code: 8152)~~

~~ERROR (Message: value too long for type character(4))~~
```

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/expected/temp_table.out [hunk 1]
```diff
@@ -63,6 +63,32 @@ int#!#varchar#!#int
 ~~END~~
 
 
+-- BABEL-5273 ALTER COLUMN to another char type
```
Comment (timchang514) [inline test/JDBC/expected/temp_table.out]: can we add some tests alter column for other types? with an empty table - like some of the other char types like nchar/nvarchar, money, datetime, int?

File: test/JDBC/expected/temp_table.out [hunk 2]
```diff
@@ -63,6 +64,176 @@ int#!#varchar#!#int
...
+ALTER TABLE #t1 ALTER COLUMN b SMALLDATETIME
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b CHAR
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b VARCHAR
+GO
+
+-- TODO: fix this, it should work and not raise a syntax error
```
Comment (timchang514) [inline test/JDBC/expected/temp_table.out:176]: should this be filed in a separate JIRA? is this temp table specific or just general issue?

File: test/JDBC/expected/temp_table.out [hunk 3]
```diff
@@ -63,6 +64,176 @@ int#!#varchar#!#int
...
+
+-- should raise error due to incompatible types
+ALTER TABLE #t1 ALTER COLUMN b GEOMETRY
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: column "b" cannot be cast automatically to type geometry)~~
+
+
+-- TODO: fix this, it should work and not raise a syntax error
```
Comment (timchang514) [inline test/JDBC/expected/temp_table.out:229]: same q as above

**Review by timchang514 — APPROVED** [review]

File: test/JDBC/expected/temp_table.out (hunk 2 above)
Comment (Sairakan) [reply test/JDBC/expected/temp_table.out:176]: I believe this is a general issue with altering table datatypes. I'll file a JIRA for this.

**Review by lejaokri — APPROVED** [review]

**Pull request merged**